org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/semantictokens/SemanticTokensVisitor.java

```diff
@@ -448,6 +456,68 @@ public boolean visit(RecordDeclaration node) {
 		return false;
 	}
 
+	// could maybe try catch unsupportedIn2
```

Comment:
What's this line indicating ? Overall it looks pretty good. So you hand the entire source to the scanner as expected but you start scanning from the type declaration being visited, because it may be nested in the type root declaration. Cool.

Also `package{Start,End}` probably should be `typeKeyword{Start,End}` or something like that.

Reply:
Ah, it was a note to myself that I missed taking out. Will update.